### DISCUSSION TITLE
Unexpected increase of packet loss rate when more RBs available

### DISCUSSION THREAD

**Discussion by zazim13** [Unipisa/Simu5G#204] [Q&A]
Hello there, 

I've been encountering unexpected performance issues with my simulations. I have implemented both a MecApp and a UEApp and I am generating traffic from an UE. I've noticed that as I increase the number of resource blocks (numBands) available, I'm unexpectedly experiencing greater packet loss. While the end-to-end delay decreases(as expected), the rate of packet loss increases.

Additionally, I have observed that my SINR decreases with more resource blocks available. 
Especially, I observed a fading attenuation 10 times higher when I increase from 200 RBs to 2000 RBs available accompanied by a rise in packet loss rate. 

This trend seems counterintuitive to me. 
I can give more details if needed.

Kind regards 
Hakim

**Comment by giovanninardini** [comment]
Hi,
what kind of traffic does the UE send? Does it send many small packets or does it send very large ones (which possibly takes up all the RBs in a scheduling interval)?

**Reply by zazim13** [reply]
Hi tanks you for your reply, 
my packet size is stochastic. It follows an exponential distribution(bounded) with an average of 40KB. Regarding the RB usage, I observe more RBs used (in active TTI) when I increase the number of RB available. 

 I see a [comment](https://github.com/Unipisa/Simu5G/blob/a28aba469a785fcd166d2fd0c41408605d86d4b6/src/stack/phy/ChannelModel/LteRealisticChannelModel.cc#L580) on considering only RB used . Can it be due to the fading noise that would be higher when there is more RB available?

**Comment by giovanninardini** [comment]
No, a random fading value is computed for each RB independently. That comment is just for improving simulation speed, i.e.: we would not need to compute fading for RBs that are not used.

I think the behavior you observed may be due to the error model used in Simu5G, which computes an error probability for each RB used for transmission. Roughly speaking, the more RB a transmission uses the higher the probability an RB is corrupted (hence, the whole transmission).

**Reply by zazim13** [reply]
I thought about that possibility 

- I indeed observe a higher probability of [error](https://github.com/Unipisa/Simu5G/blob/a28aba469a785fcd166d2fd0c41408605d86d4b6/src/stack/phy/ChannelModel/LteRealisticChannelModel.cc#L1973).

Here the average with 200 RB available
![200](https://github.com/Unipisa/Simu5G/assets/60358257/7dafb868-e67a-4772-8dd6-a207a6616e99)
Here the average with 2000 RB available
![2000](https://github.com/Unipisa/Simu5G/assets/60358257/9d0a73c1-1113-4765-acf0-b741f1d4f5d4)
The SINR is also lower with 2000 RBs than with 200 RBs
- but does it mean that for a given packet size, more or less RB will be allocated depending on the number of RBs available? If not, the increase in the number of RB used in active TTI may be due to the fact that more packets are served during each TTI or active TTI(which results in the lower RTT that I observe). 
- For some other experiments, I did 100 repetitions, and I still have more loss with more RB available. I was expecting to "avoid" a potential impact of the stochasticity of my packet size.

**Comment by giovanninardini** [comment]
> * but does it mean that for a given packet size, more or less RB will be allocated depending on the number of RBs available? If not, the increase in the number of RB used in active TTI may be due to the fact that more packets are served during each TTI or active TTI(which results in the lower RTT that I observe).

No, the number of allocated RBs depend on the buffer size and CQI only. However, assume you send a 40KB packet, it might happen that this requires allocating all the available RBs (say, 200RBs) in one TTI, maybe requiring to use also subsequent TTIs to send the whole content of the buffer. In this case the error model will compute the error probability over 200 RBs.
Now, when you increase the available bw to 2000RBs: in that case, you might use X > 200 RBs to send the whole content of the buffer - up to 2000 RBs. So, the error model will compute the error probability over such X > 200 RBs, which might be higher.

**Comment by zazim13** [comment]
Hi thank you for your reply. It seems to answer my concerns. 
However, if I've understood correctly, the probability of error remains the product of the probabilities(per RB), even when the packet is sent over subsequent Transmission Time Intervals (TTIs). The difference is that when there is more RBs used to send the packet(for instance if there is more RBs available) , the probabilities(per RB) are lower. Am I right ?  and more importantly is it the expected behavior ?

**Comment by giovanninardini** [comment]
You are correct.

It is the expected behavior for the model we decided to implement - the idea is that if there is one RB with very bad quality the frame cannot be decoded correctly (even if the average quality over the used RBs is good) because you would miss one part of the frame.
Honestly I do not know if it is the expected behavior _in general_ (I am not a PHY layer expert). However, it is quite easy to change it with a different one, for example computing the mean of the SINR over the used RBs and obtain an overall error probability from that average SINR. In case you want to do it, the code to be modified is in the `isError()` function of the LteRealisticChannelModel or NrChannelModel class

**Comment by zazim13** [comment]
Thank you for your answers and suggestions.
 I'm not an expert in the PHY layer neither, but I would be interested to hear the perspective of someone with expertise in this area.

Best regards,
Hakim
